app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -139,6 +139,11 @@ def create_vm(args = {})
       raise message
     end
 
+    def vm_ready(vm)
+      vm.wait_for { self.ready? || self.failed? }
+      raise "Failed to deploy vm %{name}, fault: %{e}" % { :name => vm.name, :e => vm.fault } if vm.failed?
```

Comment:
Please use 

```
raise Foreman::Exception.new(N_("Failed to deploy vm %{name}, fault: %{e}"), { :name => vm.name, :e => vm.fault })
```

This generates an [ERF-Code](http://projects.theforeman.org/projects/foreman/wiki/ErrorCodes) that helps with troubleshooting.